### PR TITLE
Small audio caching optimisation

### PR DESCRIPTION
**Pull request by ThomasWilshaw**
A couple of changes based of profiling while caching. An example of the improvement form the first commit:
image

The second commit gave a similar improvement.

### CODE REVIEW

File: app/render/audioparams.h [hunk 1]
```diff
@@ -75,6 +78,9 @@ class AudioParams {
   {
     set_default_footage_parameters();
     timebase_ = sample_rate_as_time_base();
+
+    // Cache channel count
+    channel_count_ = av_get_channel_layout_nb_channels(this->channel_layout());
```
Comment (itsmattkc) [inline app/render/audioparams.h:83]: You'll also want to set this under `set_channel_layout()`, and it'd probably be best to put this under a new function like `calculate_channel_count` to minimize on copy/pasted code (see `VideoParams:: calculate_effective_size`).

File: app/render/audioplaybackcache.cpp [hunk 2]
```diff
@@ -64,10 +64,14 @@ void AudioPlaybackCache::WritePCM(const TimeRange &range, SampleBufferPtr sample
...
-    playlist_.push_back(CreateSegment(seg_sz, playlist_.GetLength()));
-    length_diff -= seg_sz;
+  qint64 whole_segments = length_diff / kDefaultSegmentSize;
+  qint64 remainder = length_diff % kDefaultSegmentSize;
+
+  for (int i = 0; i < whole_segments; i++) {
+    playlist_.push_back(CreateSegment(kDefaultSegmentSize, playlist_.GetLength()));
+  }
+  if (remainder > 0) {
+    playlist_.push_back(CreateSegment(remainder, playlist_.GetLength()));
```
Comment (itsmattkc) [inline app/render/audioplaybackcache.cpp:74]: Does this actually make a difference? It looks like it does the exact same thing but with more lines of code.

**Review by itsmattkc — CHANGES_REQUESTED** [review]

File: app/render/audioparams.h (hunk 1 above)
Comment (ThomasWilshaw) [reply app/render/audioparams.h:83]: Will do, thanks.

File: app/render/audioplaybackcache.cpp (hunk 2 above)
Comment (ThomasWilshaw) [reply app/render/audioplaybackcache.cpp:74]: It did seem too, I'll try and redo the tests and post them here. As I understand it, this removes the repeated qmin and probably allows a bit of branch prediction with the if statement. Happy to be corrected though :)

Comment (itsmattkc) [reply app/render/audioplaybackcache.cpp:74]: Hmm perhaps. Yes I'll be interested to see the test information.

Comment (ThomasWilshaw) [reply app/render/audioplaybackcache.cpp:74]: I can't actually get a repeat test so I suspect it was a fluke, shall I revert the commit?

**Pull request merged**